app/controllers/users_controller.rb

```diff
@@ -31,20 +32,45 @@ def edit
   # POST /users.json
   def create
     @user = User.new(user_params)
-    user = @user
     @user.profile_picture.attach(user_params[:profile_picture])
-    if @user.username === ""
-      @user.username = @user.name + rand((User.all.length)..(User.all.length * 100000000)).to_s
-      @user.save
+    if @user.username == ""
+      while User.find_by(username: @user.username) && @user.username == ""
+        @user.username = @user.name + rand((User.all.length)..(User.all.length * 100000000)).to_s
+        @user.save
```

Comment:
Why are we using so many `@user.save`? I see one in lines 39, 45, 51, 60, and 63. I believe we just need the one in line 63 to make sure we can save the users to the db without any errors.